CI run against latest deps is failing

See https://github.com/matrix-org/synapse/actions/runs/2237655100

```
Run poetry run mypy
  poetry run mypy
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.10.4/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.4/x64/lib
synapse/logging/opentracing.py:234: error: Unused "type: ignore" comment
tests/replication/_base.py:39: error: Unused "type: ignore" comment
tests/logging/test_opentracing.py:37: error: Unused "type: ignore" comment
tests/logging/test_opentracing.py:46: error: Unused "type: ignore" comment
Found 4 errors in 3 files (checked 710 source files)
Error: Process completed with exit code 1.
```

Casualty of #12531. I think we ought to disable the unused type ignore check in this job, and probably in the twisted trunk job too.

--- a/.github/workflows/latest_deps.yml
+++ b/.github/workflows/latest_deps.yml
@@ -32,6 +32,7 @@ jobs:
         with:
           python-version: "3.x"
           poetry-version: "1.2.0b1"
+          extras: "all"
       # Dump installed versions for debugging.
       - run: poetry run pip list > before.txt
       # Upgrade all runtime dependencies only. This is intended to mimic a fresh

--- a/.github/workflows/latest_deps.yml
+++ b/.github/workflows/latest_deps.yml
@@ -38,6 +38,8 @@ jobs:
       # `pip install matrix-synapse[all]` as closely as possible.
       - run: poetry update --no-dev
       - run: poetry run pip list > after.txt && (diff -u before.txt after.txt || true)
+      - name: Remove warn_unused_ignores from mypy config
+        run: sed '/warn_unused_ignores = True/d' -i mypy.ini
       - run: poetry run mypy
   trial:
     runs-on: ubuntu-latest

--- a/.github/workflows/twisted_trunk.yml
+++ b/.github/workflows/twisted_trunk.yml
@@ -24,6 +24,8 @@ jobs:
           poetry remove twisted
           poetry add --extras tls git+https://github.com/twisted/twisted.git#trunk
           poetry install --no-interaction --extras "all test"
+      - name: Remove warn_unused_ignores from mypy config
+        run: sed '/warn_unused_ignores = True/d' -i mypy.ini
       - run: poetry run mypy
 
   trial:

--- a/changelog.d/12576.misc
+++ b/changelog.d/12576.misc
@@ -1,0 +1,1 @@
+Allow unused `#type: ignore` comments in bleeding edge CI jobs.

